src/test/java/io/confluent/examples/streams/microservices/util/MicroserviceTestUtils.java

```diff
@@ -222,4 +225,35 @@ public static void sendInventory(List<KeyValue<Product, Integer>> inventory,
       e.printStackTrace();
     }
   }
+
+  public static <T> T getWithRetries(final Invocation.Builder builder,
+                                     final GenericType<T> genericType,
+                                     int numberOfRetries) {
+    while (true) {
+      try {
+        return builder.get(genericType);
+      } catch (ServerErrorException exception) {
+        if ("HTTP 504 Gateway Timeout".equals(exception.getMessage()) && numberOfRetries-- > 0) {
+          continue;
+        }
+        throw exception;
+      }
+    }
+  }
+
+    public static <T> T getWithRetries(final Invocation.Builder builder,
+                                       final Class<T> clazz,
+                                       int numberOfRetries) {
+        while (true) {
+            try {
+                return builder.get(clazz);
+            } catch (ServerErrorException exception) {
+                if ("HTTP 504 Gateway Timeout".equals(exception.getMessage()) && numberOfRetries-- > 0) {
```

Comment:
I'm wondering if we can relax the check a bit to, say, `exception.getMessage().contains("504")`? I'm not expert in http get / put APIs but it seems various error messages may be returned: https://www.lifewire.com/504-gateway-timeout-error-explained-2622941

Reply:
Sure.